tests/integration/pilot/revision_view_test.go

```diff
@@ -0,0 +1,157 @@
+// +build integ
+//  Copyright Istio Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package pilot
+
+import (
+	"encoding/json"
+	"istio.io/istio/istioctl/cmd"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/echoboot"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+)
+
+func TestRevisionListing(t *testing.T) {
```

Comment:
we should really test the pod listing functionality, since there's a good chance of regressions for that.

Reply:
I didn't get it. Did you mean listing control plane and gateway pods when `-v` is added? Or in `revision describe -v` we show the list of pods pointing to a particular revision. That one?

Reply:
Added integration test for that case. Yes. Pod listing is a bit tricky - especially if the cluster is not in steady state. Currently, I have added 1 minute delay after setup for the cluster to be in steady state. Is there a better way?